Update ZK Stack Quickstart (#899)
matter-labs/zksync-web-era-docs: docs/zk-stack/running-a-hyperchain/locally.md

      content: Deploying Locally | zkSync Docs
---

# Getting Started with ZK Stack

## Development dependencies

> Ensure you have followed <a href="https://github.com/matter-labs/zksync-era/blob/main/docs/guides/setup-dev.md" target="_blank">these instructions</a> to set up dependencies on your machine (don't worry about the Environment section for now).

Reviewer: We should be consistent and use the square bracket notation (also helps with automatic dead links verification)
```suggestion
Ensure you have followed [these instructions](https://github.com/matter-labs/zksync-era/blob/main/docs/guides/setup-dev.md) to set up dependencies on your machine (don't worry about the Environment section for now).
```

Reviewer: Be sure to update all links on this page (I want to avoid adding duplicate comments in this PR)

Author: Updated all

4. Last, start the wizard and follow instructions to set up and deploy your new hyperchain by running `zk stack init`

   - Initially you want to `Configure new chain`

   - Give it a name and chain id.

   - Select localhost (default `matterlabs/geth`) and follow the wizard.

     - If you are doing this for the first time, several components need to be compiled/built, so do not worry if it takes a few minutes. The console will show what is going on anyways.

   - If you don't want to configure any values for now and just want check the build process for a hyperchain, try out the `zk stack demo` command.

:::warning

> The commands above are not just running docker containers, but are actually building the code from the repo to spin up your hyperchain. For this reason the process might take some time. If you just want to run docker containers to play around with a zkSync chain, you can use zksync-cli dev. Learn more here.

Reviewer 2: missing a link in "Learn more here"

Author: Thanks, fixed 👍

> The commands above are not just running docker containers, but are actually building the code from the repo to spin up your hyperchain. For this reason the process might take some time. If you just want to run docker containers to play around with a zkSync chain, you can use zksync-cli dev. Learn more here.

Reviewer 2: ```suggestion
The commands above are not just running docker containers, but are actually building the code from the repo to spin up your hyperchain. For this reason the process might take some time. If you just want to run docker containers to play around with a zkSync chain, you can use `npx zksync-cli dev`. Learn more here.
```

:::

Your hyperchain is now deployed to the base chain (most likely a local geth docker container) and configured. You can find all configuration in a new `.env` file created on `<project root>/etc/env/<your_chain_name_slug>.env`, and if you deployed test tokens, their addresses will be available at `<project root>/etc/tokens/<the_l1_identifier>.json`

1. The wizard allows you to run the server in the end. If you chose not to, you’re still able to run it by executing

```bash
zk server --components "http_api,eth,state_keeper,housekeeper"
```

2. You can now run transactions and start playing with your hyperchain by using the RPC available at<http://localhost:3050>.

>    - Don't forget to deposit some ETH and fund your accounts on your hyperchain. To do so follow these instructions ->

Reviewer: Is there content missing to the right of the arrow?

Reviewer: Perhaps it's just confusing to see these arrows at the end of sentences

Author: This came from weird formatting on the quickstart. Updated to be hyperlink. Alternatively can also just remove and make "Follow the instructions below".

## Using your hyperchain

During the `zk stack init` configurator, you have a choice of what base layer to deploy the hyperchain onto: local geth node, or an Ethereum testnet. The first step to start interacting with your hyperchain is to fund an account (or a few). This means you need some funds on the base layer ->

> ### Base layer is the local geth node (<a href="https://hub.docker.com/r/matterlabs/geth" target="_blank">@matterlabs/geth:latest</a>)

Reviewer: ```suggestion
### Base layer is the local geth node ([@matterlabs/geth:latest](https://hub.docker.com/r/matterlabs/geth))
```

> You will have a set of addresses that have 100 ETH each. You can find the list here and use these addresses to deposit into your hyperchain.

Author: Yupp thanks updated

### Base layer is an Ethereum network (e.g., Goerli, Sepolia)

In this case, you need to have an account on the base layer with ETH (which, for sure, you have given the deployer, governor, and operator needed ETH to go through the deployment process). You can use any of these wallets (or any other you have funds) to deposit into your hyperchain.

> Once you have the accounts with funds on the base layer, you can do a deposit (and any further interactions) in 3 ways ->

Reviewer 2: not sure what the "->" means here. If it's just pointing the 3 different ways to use the hyperchain RPC mentioned below, I'd be specific about it

Author: Changed back to ":"

## Using your hyperchain RPC

> Your server contains both HTTPS as well as WebSocket (WS) services that are fully web3 compatible (and contain some extra ZK Stack functionalities). Learn more about it <a href="https://docs.zksync.io/build/api.html" target="_blank">here</a>.

Reviewer: This should link directly to the `api.md` page
```suggestion
Your server contains both HTTPS as well as WebSocket (WS) services that are fully web3 compatible (and contain some extra ZK Stack functionalities). Learn more about it [here](../../build/api.md).
```

- 64 GB of RAM
- 300 GB of Disk Space (SSD preferred)

### Requirements for <u>CPU</u> Prover

The docker compose file assumes you will be running all components in the same machine. The current minimum requirements for a low TPS scenario are:

- 32 Core CPU
- 128 GB of RAM
- 700 of Disk Space (SSD preferred)

## Addendum

- If you make changes to any contract, you can always deploy a new hyperchain to easily test those changes.

- If you configure your hyperchain once, you don't need to do it again as the wizard allows you to use an existing config file.

- For now, it is only possible to deploy a hyperchain as an L2, but soon it will also work as L3s.

- When running the default matterlabs/geth, you have a set of rich wallets available to you. You can check them <a href="https://github.com/matter-labs/local-setup/blob/main/rich-wallets.json" target="_blank">here</a>.

- If you want to have a custom local base chain, you must ensure you have a database for your hyperchain, as well as the local RPC for your L1.

  - To run a Postgres 14 database for your hyperchain, execute the following:

```bash
docker-compose -f docker-compose-zkstack-common.yml up -d postgres
```

> in case you don't want to use the docker Postgres database above but another one you already have locally, make sure its version is 14 and it is running and accepts connections at `postgres://postgres@localhost/zksync_local`. You can test with:

Reviewer: nit: Uppercase start of sentence
```suggestion
In case you don't want to use the docker Postgres database above but another one you already have locally, make sure its version is 14 and it is running and accepts connections at `postgres://postgres@localhost/zksync_local`. You can test with:
```
